// antd-mobile/lib/switch/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports["default"] = undefined;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _reactNative = require('react-native');

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj["default"] = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var AntmSwitch = function (_React$Component) {
    (0, _inherits3["default"])(AntmSwitch, _React$Component);

    function AntmSwitch(props) {
        (0, _classCallCheck3["default"])(this, AntmSwitch);

        var _this = (0, _possibleConstructorReturn3["default"])(this, _React$Component.call(this, props));

        _this.state = {
            checked: props.checked
        };
        return _this;
    }

    AntmSwitch.prototype.onChange = function onChange(value) {
        this.props.onChange(value);
        this.setState({
            checked: value
        });
    };

    AntmSwitch.prototype.render = function render() {
        var _this2 = this;

        var _props = this.props;
        var style = _props.style;
        var disabled = _props.disabled;

        return React.createElement(_reactNative.Switch, { style: style, onValueChange: function onValueChange(value) {
                _this2.onChange(value);
            }, value: this.state.checked, disabled: disabled });
    };

    return AntmSwitch;
}(React.Component);

exports["default"] = AntmSwitch;

AntmSwitch.propTypes = {
    style: _react.PropTypes.object,
    checked: _react.PropTypes.bool,
    disabled: _react.PropTypes.bool,
    onChange: _react.PropTypes.func
};
AntmSwitch.defaultProps = {
    name: '',
    checked: false,
    disabled: false,
    onChange: function onChange() {}
};
module.exports = exports['default'];
