// antd-mobile/lib/refresh-control/index.web.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports["default"] = undefined;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _rmcPullToRefresh = require('rmc-pull-to-refresh');

var _rmcPullToRefresh2 = _interopRequireDefault(_rmcPullToRefresh);

var _icon = require('../icon');

var _icon2 = _interopRequireDefault(_icon);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj["default"] = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var RefreshControl = function (_React$Component) {
    (0, _inherits3["default"])(RefreshControl, _React$Component);

    function RefreshControl() {
        (0, _classCallCheck3["default"])(this, RefreshControl);
        return (0, _possibleConstructorReturn3["default"])(this, _React$Component.apply(this, arguments));
    }

    RefreshControl.prototype.render = function render() {
        return React.createElement(_rmcPullToRefresh2["default"], this.props);
    };

    return RefreshControl;
}(React.Component);

exports["default"] = RefreshControl;

RefreshControl.propTypes = {
    prefixCls: _react.PropTypes.string,
    children: _react.PropTypes.any,
    icon: _react.PropTypes.any,
    loading: _react.PropTypes.any,
    scrollerOptions: _react.PropTypes.object
};
RefreshControl.defaultProps = {
    prefixCls: 'am-refresh-control',
    icon: React.createElement(
        'div',
        { style: { lineHeight: '50px' } },
        React.createElement(
            'div',
            { className: 'am-refresh-control-pull' },
            React.createElement(_icon2["default"], { type: 'arrow-down' }),
            ' 下拉'
        ),
        React.createElement(
            'div',
            { className: 'am-refresh-control-release' },
            React.createElement(_icon2["default"], { type: 'arrow-up' }),
            ' 释放'
        )
    ),
    loading: React.createElement(
        'div',
        { style: { lineHeight: '50px' } },
        React.createElement(_icon2["default"], { type: 'loading' })
    )
};
module.exports = exports['default'];
