// antd-mobile/lib/steps/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports["default"] = undefined;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _StepsItem = require('./StepsItem');

var _StepsItem2 = _interopRequireDefault(_StepsItem);

var _reactNative = require('react-native');

var _style = require('./style');

var _style2 = _interopRequireDefault(_style);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj["default"] = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Steps = function (_React$Component) {
    (0, _inherits3["default"])(Steps, _React$Component);

    function Steps(props) {
        (0, _classCallCheck3["default"])(this, Steps);

        var _this = (0, _possibleConstructorReturn3["default"])(this, _React$Component.call(this, props));

        _this.onLayout = function (e) {
            _this.setState({
                wrapWidth: e.nativeEvent.layout.width
            });
        };
        _this.state = {
            wrapWidth: 0
        };
        return _this;
    }

    Steps.prototype.render = function render() {
        var _this2 = this;

        var children = this.props.children;
        var wrapView = this.props.direction === 'vertical' ? null : 'warp_row';
        return React.createElement(
            _reactNative.View,
            { style: _style2["default"][wrapView], onLayout: function onLayout(e) {
                    _this2.onLayout(e);
                } },
            React.Children.map(children, function (ele, idx) {
                var errorTail = -1;
                if (idx < children.length - 1) {
                    var status = _this2.props.children[idx + 1].props.status;
                    if (status === 'error') {
                        errorTail = idx;
                    }
                }
                return React.cloneElement(ele, {
                    index: idx,
                    last: idx === children.length - 1,
                    direction: _this2.props.direction,
                    current: _this2.props.current,
                    width: 1 / (children.length - 1) * _this2.state.wrapWidth,
                    size: _this2.props.size,
                    finishIcon: _this2.props.finishIcon,
                    errorTail: errorTail
                });
            })
        );
    };

    return Steps;
}(React.Component);

exports["default"] = Steps;

Steps.defaultProps = {
    direction: ''
};
Steps.Step = _StepsItem2["default"];
module.exports = exports['default'];
