// antd-mobile/lib/steps/index.web.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports["default"] = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _rcSteps = require('rc-steps');

var _rcSteps2 = _interopRequireDefault(_rcSteps);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj["default"] = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Steps = function (_React$Component) {
    (0, _inherits3["default"])(Steps, _React$Component);

    function Steps() {
        (0, _classCallCheck3["default"])(this, Steps);
        return (0, _possibleConstructorReturn3["default"])(this, _React$Component.apply(this, arguments));
    }

    Steps.prototype.render = function render() {
        var _this2 = this;

        var _props = this.props;
        var current = _props.current;
        var direction = _props.direction;

        return React.createElement(
            _rcSteps2["default"],
            (0, _extends3["default"])({}, this.props, { direction: direction }),
            this.props.children.map(function (item, index) {
                var errorTail = -1;
                if (index < _this2.props.children.length - 1) {
                    var status = _this2.props.children[index + 1].props.status;
                    if (status === 'error') {
                        errorTail = index;
                    }
                }
                var errorTailCls = errorTail > -1 ? 'error-tail' : '';
                var iconName = void 0;
                var className = void 0;
                if (!!item.props.icon) {
                    iconName = item.props.icon;
                    className = '';
                    if (index > 0 && index <= current) {
                        iconName = 'check-circle';
                    } else if (item.props.status === 'error') {
                        iconName = 'cross-circle';
                    } else if (item.props.status === 'process') {
                        iconName = 'check-circle';
                    }
                } else {
                    className = index <= current ? null : 'ellipsis-item';
                    if (index <= current) {
                        iconName = 'check-circle-o';
                    } else if (item.props.status === 'error') {
                        iconName = 'cross-circle-o';
                    } else {
                        iconName = 'ellipsis';
                    }
                }
                className = errorTailCls + ' ' + className;
                return React.cloneElement(item, { key: index, icon: iconName, className: className });
            })
        );
    };

    return Steps;
}(React.Component);

exports["default"] = Steps;

Steps.Step = _rcSteps2["default"].Step;
Steps.defaultProps = {
    prefixCls: 'am-steps',
    iconPrefix: 'ant',
    labelPlacement: 'vertical',
    current: 0,
    direction: 'vertical'
};
module.exports = exports['default'];
