// antd-mobile/lib/tabs/index.web.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _rcTabs = require('rc-tabs');

var _rcTabs2 = _interopRequireDefault(_rcTabs);

var _SwipeableTabContent = require('rc-tabs/lib/SwipeableTabContent');

var _SwipeableTabContent2 = _interopRequireDefault(_SwipeableTabContent);

var _TabContent = require('rc-tabs/lib/TabContent');

var _TabContent2 = _interopRequireDefault(_TabContent);

var _InkTabBar = require('rc-tabs/lib/InkTabBar');

var _InkTabBar2 = _interopRequireDefault(_InkTabBar);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj["default"] = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Tabs = React.createClass({
    displayName: 'Tabs',

    statics: {
        TabPane: _rcTabs.TabPane
    },
    getDefaultProps: function getDefaultProps() {
        return {
            prefixCls: 'am-tabs',
            animated: true,
            swipeable: true,
            onChange: function onChange() {},

            tabBarPosition: 'top',
            onTabClick: function onTabClick() {}
        };
    },
    renderTabBar: function renderTabBar() {
        var props = this.props;

        return React.createElement(_InkTabBar2["default"], { onTabClick: props.onTabClick, inkBarAnimated: props.animated });
    },
    renderTabContent: function renderTabContent() {
        var _props = this.props;
        var animated = _props.animated;
        var swipeable = _props.swipeable;

        return swipeable ? React.createElement(_SwipeableTabContent2["default"], { animated: animated }) : React.createElement(_TabContent2["default"], { animated: animated });
    },
    render: function render() {
        return React.createElement(_rcTabs2["default"], (0, _extends3["default"])({ renderTabBar: this.renderTabBar, renderTabContent: this.renderTabContent }, this.props));
    }
});
exports["default"] = Tabs;
module.exports = exports['default'];
