// antd-mobile/lib/modal/style/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _default = require('../../style/themes/default');

var _default2 = _interopRequireDefault(_default);

var _reactNative = require('react-native');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

exports["default"] = _reactNative.StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: _default2["default"].fill_overlay_inverse
    },
    innerContainer: {
        borderRadius: _default2["default"].radius_md,
        width: 286,
        paddingTop: _default2["default"].v_spacing_xl,
        overflow: 'hidden'
    },
    header: {
        fontSize: _default2["default"].font_size_heading,
        color: _default2["default"].color_text_base,
        textAlign: 'center',
        paddingHorizontal: _default2["default"].h_spacing_lg
    },
    body: {
        paddingTop: 0,
        paddingBottom: _default2["default"].v_spacing_lg,
        paddingHorizontal: _default2["default"].h_spacing_lg
    },
    maskClosable: {
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
        backgroundColor: 'transparent'
    },
    closeWrap: {
        position: 'absolute',
        top: 0,
        left: _default2["default"].h_spacing_lg
    },
    close: {
        fontSize: 40,
        fontWeight: '200',
        color: '#bcbcbc'
    },
    buttnGroupH: {
        flexDirection: 'row',
        borderTopWidth: _default2["default"].border_width_sm,
        borderColor: _default2["default"].border_color_base
    },
    buttnGroupV: {
        flexDirection: 'column'
    },
    buttnWrapH: {
        flex: 1,
        borderColor: _default2["default"].border_color_base,
        borderRightWidth: _default2["default"].border_width_sm,
        paddingVertical: 11
    },
    buttnWrapV: {
        flex: 1,
        borderTopWidth: _default2["default"].border_width_sm,
        borderColor: _default2["default"].border_color_base,
        paddingVertical: 11
    },
    buttonText: {
        textAlign: 'center',
        color: _default2["default"].color_link,
        fontSize: _default2["default"].link_button_font_size
    }
});
module.exports = exports['default'];
