// rmc-date-picker/lib/DatePicker.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _reactNative = require('react-native');

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _Picker = require('rmc-picker/lib/Picker');

var _Picker2 = _interopRequireDefault(_Picker);

var _DatePickerMixin = require('./DatePickerMixin');

var _DatePickerMixin2 = _interopRequireDefault(_DatePickerMixin);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj["default"] = obj; return newObj; } }

var styles = _reactNative.StyleSheet.create({
    item: {
        flex: 1
    },
    root: {
        flexDirection: 'row',
        alignItems: 'center',
        paddingTop: 10,
        paddingBottom: 10
    }
});
var DatePicker = React.createClass({
    displayName: 'DatePicker',

    mixins: [_DatePickerMixin2["default"]],
    render: function render() {
        var _this = this;

        var _getValueDataSource = this.getValueDataSource();

        var value = _getValueDataSource.value;
        var dataSource = _getValueDataSource.dataSource;

        var inner = dataSource.map(function (items, i) {
            return React.createElement(
                _reactNative.View,
                { style: styles.item, key: i },
                React.createElement(
                    _Picker2["default"],
                    { pure: false, selectedValue: value[i], onValueChange: function onValueChange(v) {
                            _this.onValueChange(i, v);
                        } },
                    items
                )
            );
        });
        return React.createElement(
            _reactNative.View,
            { style: styles.root },
            inner
        );
    }
});
exports["default"] = DatePicker;
module.exports = exports['default'];
