// antd-mobile/lib/carousel/index.web.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports["default"] = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _nukaCarousel = require('nuka-carousel');

var _nukaCarousel2 = _interopRequireDefault(_nukaCarousel);

var _objectAssign = require('object-assign');

var _objectAssign2 = _interopRequireDefault(_objectAssign);

var _index = require('../pagination/index.web');

var _index2 = _interopRequireDefault(_index);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj["default"] = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Carousel = function (_React$Component) {
    (0, _inherits3["default"])(Carousel, _React$Component);

    function Carousel(props) {
        (0, _classCallCheck3["default"])(this, Carousel);

        var _this = (0, _possibleConstructorReturn3["default"])(this, _React$Component.call(this, props));

        _this.state = {
            selectedIndex: _this.props.selectedIndex
        };
        _this.onChange = _this.onChange.bind(_this);
        return _this;
    }

    Carousel.prototype.onChange = function onChange(index) {
        this.setState({ selectedIndex: index });
    };

    Carousel.prototype.renderDots = function renderDots() {
        return React.createElement(_index2["default"], { mode: 'pointer', current: this.state.selectedIndex, total: this.props.children.length });
    };

    Carousel.prototype.render = function render() {
        var children = this.props.children;
        if (!children) {
            return null;
        }
        var props = (0, _objectAssign2["default"])({}, this.props);
        if (props.infinite) {
            props.wrapAround = true;
        }
        if (props.selectedIndex) {
            props.slideIndex = props.selectedIndex;
        }
        if (props.beforeChange) {
            props.beforeSlide = props.beforeChange;
        }
        if (props.afterChange) {
            props.afterSlide = props.afterChange;
        }
        var className = 'am-carousel';
        if (props.vertical) {
            className = className + ' am-carousel-vertical';
        }
        return React.createElement(
            'div',
            { className: className },
            React.createElement(_nukaCarousel2["default"], (0, _extends3["default"])({}, props, { decorators: [], afterSlide: this.onChange })),
            props.dots && this.renderDots()
        );
    };

    return Carousel;
}(React.Component);

exports["default"] = Carousel;

Carousel.defaultProps = {
    dots: true,
    arrows: false,
    autoplay: false,
    infinite: false,
    edgeEasing: 'linear',
    cellAlign: 'center',
    selectedIndex: 0
};
module.exports = exports['default'];
