// antd-mobile/lib/flex/Flex.web.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports["default"] = undefined;

var _defineProperty2 = require('babel-runtime/helpers/defineProperty');

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj["default"] = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Flex = function (_React$Component) {
    (0, _inherits3["default"])(Flex, _React$Component);

    function Flex() {
        (0, _classCallCheck3["default"])(this, Flex);
        return (0, _possibleConstructorReturn3["default"])(this, _React$Component.apply(this, arguments));
    }

    Flex.prototype.render = function render() {
        var _classNames;

        var _props = this.props;
        var direction = _props.direction;
        var wrap = _props.wrap;
        var justify = _props.justify;
        var align = _props.align;
        var alignContent = _props.alignContent;
        var className = _props.className;
        var children = _props.children;
        var prefixCls = _props.prefixCls;
        var style = _props.style;
        var onClick = _props.onClick;

        var wrapCls = (0, _classnames2["default"])((_classNames = {}, (0, _defineProperty3["default"])(_classNames, prefixCls, true), (0, _defineProperty3["default"])(_classNames, prefixCls + '-dir-row', direction === 'row'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-dir-row-reverse', direction === 'row-reverse'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-dir-column', direction === 'column'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-dir-column-reverse', direction === 'column-reverse'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-nowrap', wrap === 'nowrap'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-wrap', wrap === 'wrap'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-wrap-reverse', wrap === 'wrap-reverse'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-justify-start', justify === 'start'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-justify-end', justify === 'end'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-justify-center', justify === 'center'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-justify-between', justify === 'between'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-justify-around', justify === 'around'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-align-top', align === 'top' || align === 'start'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-align-middle', align === 'middle' || align === 'center'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-align-bottom', align === 'bottom' || align === 'end'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-align-baseline', align === 'baseline'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-align-stretch', align === 'stretch'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-align-content-start', alignContent === 'start'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-align-content-end', alignContent === 'end'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-align-content-center', alignContent === 'center'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-align-content-between', alignContent === 'between'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-align-content-around', alignContent === 'around'), (0, _defineProperty3["default"])(_classNames, prefixCls + '-align-content-stretch', alignContent === 'stretch'), (0, _defineProperty3["default"])(_classNames, className, className), _classNames));
        return React.createElement(
            'div',
            { className: wrapCls, style: style, onClick: onClick },
            children
        );
    };

    return Flex;
}(React.Component);

exports["default"] = Flex;

Flex.propTypes = {
    direction: _react.PropTypes.oneOf(['row', 'row-reverse', 'column', 'column-reverse']),
    wrap: _react.PropTypes.oneOf(['nowrap', 'wrap', 'wrap-reverse']),
    justify: _react.PropTypes.oneOf(['start', 'end', 'center', 'between', 'around']),
    align: _react.PropTypes.oneOf(['start', 'end', 'center', 'top', 'middle', 'bottom', 'baseline', 'stretch']),
    alignContent: _react.PropTypes.oneOf(['start', 'end', 'center', 'between', 'around', 'stretch'])
};
Flex.defaultProps = {
    prefixCls: 'am-flexbox',
    align: 'center',
    onClick: function onClick() {}
};
module.exports = exports['default'];
