// antd-mobile/lib/tag/style/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _default = require('../../style/themes/default');

var _default2 = _interopRequireDefault(_default);

var _reactNative = require('react-native');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

exports["default"] = _reactNative.StyleSheet.create({
    tag: {
        borderRadius: _default2["default"].radius_sm,
        backgroundColor: 'transparent',
        flexDirection: 'row',
        overflow: 'hidden'
    },
    wrap: {
        overflow: 'hidden',
        borderRadius: _default2["default"].radius_sm,
        borderWidth: _default2["default"].border_width_md,
        borderStyle: 'solid',
        paddingVertical: _default2["default"].v_spacing_sm,
        paddingHorizontal: _default2["default"].h_spacing_lg
    },
    text: {
        fontSize: _default2["default"].button_font_size_sm,
        minWidth: 80,
        textAlign: 'center'
    },
    normalWrap: {
        backgroundColor: _default2["default"].fill_base,
        borderColor: _default2["default"].border_color_base
    },
    normalText: {
        color: _default2["default"].color_text_caption
    },
    activeWrap: {
        backgroundColor: _default2["default"].fill_base,
        borderColor: _default2["default"].brand_primary
    },
    activeText: {
        color: _default2["default"].color_link
    },
    disabledWrap: {
        backgroundColor: _default2["default"].fill_disabled,
        borderWidth: 0
    },
    disabledText: {
        color: _default2["default"].color_text_disabled
    }
});
module.exports = exports['default'];
