// antd-mobile/lib/search-bar/SearchBarPropTypes.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.defaultProps = exports.propTypes = undefined;

var _react = require('react');

function noop() {}
var propTypes = exports.propTypes = {
    prefixCls: _react.PropTypes.string,
    style: _react.PropTypes.object,
    defaultValue: _react.PropTypes.string,
    value: _react.PropTypes.string,
    placeholder: _react.PropTypes.string,
    onSubmit: _react.PropTypes.func,
    onChange: _react.PropTypes.func,
    onFocus: _react.PropTypes.func,
    onBlur: _react.PropTypes.func,
    onCancel: _react.PropTypes.func,
    onClear: _react.PropTypes.func,
    showCancelButton: _react.PropTypes.bool,
    cancelText: _react.PropTypes.string,
    disabled: _react.PropTypes.bool
};
var defaultProps = exports.defaultProps = {
    prefixCls: 'am-search',
    placeholder: '',
    onSubmit: noop,
    onChange: noop,
    onFocus: noop,
    onBlur: noop,
    onClear: noop,
    showCancelButton: false,
    cancelText: '取消',
    disabled: false
};
