// antd-mobile/lib/drawer/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports["default"] = undefined;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _slicedToArray2 = require('babel-runtime/helpers/slicedToArray');

var _slicedToArray3 = _interopRequireDefault(_slicedToArray2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _reactNative = require('react-native');

var _reactNativeDrawerLayout = require('react-native-drawer-layout');

var _reactNativeDrawerLayout2 = _interopRequireDefault(_reactNativeDrawerLayout);

var _splitObject3 = require('../_util/splitObject');

var _splitObject4 = _interopRequireDefault(_splitObject3);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj["default"] = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Drawer = function (_React$Component) {
    (0, _inherits3["default"])(Drawer, _React$Component);

    function Drawer() {
        (0, _classCallCheck3["default"])(this, Drawer);
        return (0, _possibleConstructorReturn3["default"])(this, _React$Component.apply(this, arguments));
    }

    Drawer.prototype.render = function render() {
        var _this2 = this;

        var _splitObject = (0, _splitObject4["default"])(this.props, ['children', 'sidebar', 'onOpenChange', 'position', 'drawerWidth', 'drawerBackgroundColor']);

        var _splitObject2 = (0, _slicedToArray3["default"])(_splitObject, 2);

        var _splitObject2$ = _splitObject2[0];
        var children = _splitObject2$.children;
        var sidebar = _splitObject2$.sidebar;
        var onOpenChange = _splitObject2$.onOpenChange;
        var position = _splitObject2$.position;
        var drawerWidth = _splitObject2$.drawerWidth;
        var drawerBackgroundColor = _splitObject2$.drawerBackgroundColor;
        var restProps = _splitObject2[1];

        if (_reactNative.Platform.OS === 'android') {
            position = _reactNativeDrawerLayout2["default"].positions.Left;
            if (position === 'right') {
                position = _reactNativeDrawerLayout2["default"].positions.Right;
            }
        }
        return React.createElement(
            _reactNativeDrawerLayout2["default"],
            (0, _extends3["default"])({ ref: function ref(drawer) {
                    return _this2.drawer = drawer;
                }, renderNavigationView: function renderNavigationView() {
                    return sidebar;
                }, drawerWidth: drawerWidth, drawerBackgroundColor: drawerBackgroundColor, drawerPosition: position, onDrawerOpen: function onDrawerOpen() {
                    return onOpenChange(true);
                }, onDrawerClose: function onDrawerClose() {
                    return onOpenChange(false);
                } }, restProps),
            children
        );
    };

    return Drawer;
}(React.Component);

exports["default"] = Drawer;

Drawer.propTypes = {
    children: _react.PropTypes.any
};
Drawer.defaultProps = {
    position: 'left',
    onOpenChange: function onOpenChange() {},
    drawerWidth: 300
};
module.exports = exports['default'];
